pkg/virt-controller/services/template.go

```diff
@@ -162,13 +164,37 @@ func (t *templateService) RenderLaunchManifest(vm *v1.VirtualMachine) (*k8sv1.Po
 	// Add memory overhead
 	setMemoryOverhead(vm.Spec.Domain, &resources)
 
+	// Read requested hookSidecars from VM annotation
+	var err error
+	requestedHookSidecarList := make(hooks.HookSidecarList, 0)
+	if rawRequestedHookSidecarList, requestedHookSidecarListDefined := vm.GetObjectMeta().GetAnnotations()[hooks.HookSidecarListAnnotationName]; requestedHookSidecarListDefined {
+		requestedHookSidecarList, err = hooks.UnmarshalHookSidecarList(rawRequestedHookSidecarList)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if len(requestedHookSidecarList) != 0 {
+		volumes = append(volumes, k8sv1.Volume{
+			Name: "hook-sidecar-sockets",
+			VolumeSource: k8sv1.VolumeSource{
+				EmptyDir: &k8sv1.EmptyDirVolumeSource{},
+			},
+		})
+		volumesMounts = append(volumesMounts, k8sv1.VolumeMount{
+			Name:      "hook-sidecar-sockets",
+			MountPath: hooks.HookSocketsSharedDirectory,
+		})
+	}
+
 	command := []string{"/entrypoint.sh",
 		"--qemu-timeout", "5m",
 		"--name", domain,
 		"--namespace", namespace,
 		"--kubevirt-share-dir", t.virtShareDir,
 		"--readiness-file", "/tmp/healthy",
 		"--grace-period-seconds", strconv.Itoa(int(gracePeriodSeconds)),
+		"--hook-sidecars", strconv.Itoa(len(requestedHookSidecarList)),
```

Comment:
wonder if these should be paths to the sockets i.e.: ` my_first_hook.sock, my_second_hook.sock ... my_N_hook.sock`

Reply:
We don't know name of the socket in `template.go`, we just except defined number of hook sidecars. That should be ok as long as one of the sidecars does not expose multiple hooks at the same time.